### PR TITLE
added checks for system-only DBus

### PR DESCRIPTION
**Pull request by Liboicl**
I run a Linux distribution that doesn't implement a session DBus, but does have a system-wide DBus. This change simply checks if the DBus connection fails and tries to connect to the system DBus. The ExceptionInInitializerError exception is also thrown by the SecretServiceKeychain because of this. I added that check to notify that it isn't supported. 
It fixes this issue for me https://github.com/cryptomator/cryptomator/issues/1479

### CODE REVIEW

File: src/main/java/org/cryptomator/linux/keychain/KDEWalletKeychainAccess.java [hunk 1]
```diff
@@ -24,7 +24,15 @@
 	public KDEWalletKeychainAccess() {
 		ConnectedWallet wallet = null;
 		try {
-			DBusConnection conn = DBusConnection.getConnection(DBusConnection.DBusBusType.SESSION);
+			DBusConnection conn = null;
+			try {
+				conn = DBusConnection.getConnection(DBusConnection.DBusBusType.SESSION);
+			} catch (RuntimeException e) {
```
Comment (purejava) [inline src/main/java/org/cryptomator/linux/keychain/KDEWalletKeychainAccess.java:30]: `DBusConnection.getConnection` throws a `DBusException` doesn't it? I'd suggest to catch that instead.

File: src/main/java/org/cryptomator/linux/keychain/SecretServiceKeychainAccess.java [hunk 2]
```diff
@@ -18,7 +18,7 @@ public boolean isSupported() {
 		try (@SuppressWarnings("unused") SimpleCollection keyring = new SimpleCollection()) {
 			// seems like we're able to access the keyring.
 			return true;
-		} catch (IOException | RuntimeException e) {
+		} catch (IOException | ExceptionInInitializerError | RuntimeException e) {
```
Comment (purejava) [inline src/main/java/org/cryptomator/linux/keychain/SecretServiceKeychainAccess.java:21]: Why did you add the `ExceptionInInitializerError` here? I think secret-service throws an `IOException` if something goes wrong?

**Review by purejava — CHANGES_REQUESTED** [review]
I tested the PR on Kubuntu 20.4 and DBus initialization does still work there.
I do not have a Linux without a DBusType.SESSION to test the PR there myself.

File: src/main/java/org/cryptomator/linux/keychain/KDEWalletKeychainAccess.java (hunk 1 above)
Comment (Liboicl) [reply src/main/java/org/cryptomator/linux/keychain/KDEWalletKeychainAccess.java:30]: It can throw a DBusException on certain errors but in the case of a non-existent session DBus it is only throwing a RuntimeException as per this line
`Caused by: java.lang.RuntimeException: Cannot Resolve Session Bus Address`
Using a DBusException in place of it results in the same error as before and the catch is never executed.
There may be a better way to check before the exception is thrown... Let me look into it a bit more.

File: src/main/java/org/cryptomator/linux/keychain/SecretServiceKeychainAccess.java (hunk 2 above)
Comment (Liboicl) [reply src/main/java/org/cryptomator/linux/keychain/SecretServiceKeychainAccess.java:21]: That's based upon the actual code running on my setup. With the Session DBus non-existent it threw an ExceptionInInitializerError upon running, so I just added it to the list so it marks it as not supported. Without that, it crashes similarly to the original issue, but with SecretService in the backtrace.

**Review by purejava — APPROVED** [review]

**Comment by Liboicl** [comment]
A follow up on the DBusException. It seems the version used in cryptomator is https://github.com/hypfvieh/dbus-java and for whatever reason, this version throws a RuntimeException instead of a DBusExcepiton when throwing "Cannot Resolve Session Bus Address"

From line 256-258 in org.freedesktop.dbus.connections.impl.DBusConnection
`                            if (!addressfile.exists()) {
                                throw new RuntimeException("Cannot Resolve Session Bus Address");
                            }`

**Comment by swiesend** [comment]
I released the secret-service 1.3.0, which should address this issue.

**Comment by Liboicl** [comment]
@swiesend I tested your pull request and it definitely didn't fix the issue on my system. I tried it straight from your repo and with only my changes to KDEWallet, but without checking for the "ExceptionInInitializerError" I still get
`14:52:39.844 [JavaFX Application Thread] INFO  o.c.ui.launcher.FxApplicationStarter - JavaFX Runtime started.
14:52:40.708 [JavaFX Application Thread] WARN  o.c.l.k.KDEWalletKeychainAccess - SESSION DBus not found.
Exception in thread "JavaFX Application Thread" java.lang.ExceptionInInitializerError
	at org.cryptomator.linux.keychain.SecretServiceKeychainAccess.isSupported(SecretServiceKeychainAccess.java:19)
	at java.base/java.util.stream.ReferencePipeline$2$1.accept(ReferencePipeline.java:176)
	at java.base/java.util.stream.ReferencePipeline$3$1.accept(ReferencePipeline.java:195)
	at java.base/java.util.Iterator.forEachRemaining(Iterator.java:133)
	at java.base/java.util.Spliterators$IteratorSpliterator.forEachRemaining(Spliterators.java:1801)
	at java.base/java.util.stream.AbstractPipeline.copyInto(AbstractPipeline.java:484)
	at java.base/java.util.stream.AbstractPipeline.wrapAndCopyInto(AbstractPipeline.java:474)
	at java.base/java.util.stream.ReduceOps$ReduceOp.evaluateSequential(ReduceOps.java:913)
	at java.base/java.util.stream.AbstractPipeline.evaluate(AbstractPipeline.java:234)
	at java.base/java.util.stream.ReferencePipeline.collect(ReferencePipeline.java:578)
	at org.cryptomator.common.keychain.KeychainModule.provideSupportedKeychainAccessProviders(KeychainModule.java:30)
	at org.cryptomator.common.keychain.KeychainModule_ProvideSupportedKeychainAccessProvidersFactory.provideSupportedKeychainAccessProviders(KeychainModule_ProvideSupportedKeychainAccessProvidersFactory.java:39)
	at org.cryptomator.common.keychain.KeychainModule_ProvideSupportedKeychainAccessProvidersFactory.get(KeychainModule_ProvideSupportedKeychainAccessProvidersFactory.java:29)
	at org.cryptomator.common.keychain.KeychainModule_ProvideSupportedKeychainAccessProvidersFactory.get(KeychainModule_ProvideSupportedKeychainAccessProvidersFactory.java:11)
	at dagger.internal.DoubleCheck.get(DoubleCheck.java:47)
	at org.cryptomator.common.keychain.KeychainModule_ProvideKeychainAccessProviderFactory.get(KeychainModule_ProvideKeychainAccessProviderFactory.java:33)
	at org.cryptomator.common.keychain.KeychainModule_ProvideKeychainAccessProviderFactory.get(KeychainModule_ProvideKeychainAccessProviderFactory.java:12)
	at dagger.internal.DoubleCheck.get(DoubleCheck.java:47)
	at org.cryptomator.common.keychain.KeychainManager_Factory.get(KeychainManager_Factory.java:27)
	at org.cryptomator.common.keychain.KeychainManager_Factory.get(KeychainManager_Factory.java:9)
	at dagger.internal.DoubleCheck.get(DoubleCheck.java:47)
	at org.cryptomator.ui.mainwindow.VaultDetailLockedController_Factory.get(VaultDetailLockedController_Factory.java:45)
	at org.cryptomator.ui.mainwindow.VaultDetailLockedController_Factory.get(VaultDetailLockedController_Factory.java:13)
	at dagger.internal.DoubleCheck.get(DoubleCheck.java:47)
	at org.cryptomator.ui.common.FXMLLoaderFactory.constructController(FXMLLoaderFactory.java:75)
	at javafx.fxml.FXMLLoader$ValueElement.processAttribute(FXMLLoader.java:940)
	at javafx.fxml.FXMLLoader$InstanceDeclarationElement.processAttribute(FXMLLoader.java:982)
	at javafx.fxml.FXMLLoader$Element.processStartElement(FXMLLoader.java:229)
	at javafx.fxml.FXMLLoader$ValueElement.processStartElement(FXMLLoader.java:754)
	at javafx.fxml.FXMLLoader.processStartElement(FXMLLoader.java:2808)
	at javafx.fxml.FXMLLoader.loadImpl(FXMLLoader.java:2634)
	at javafx.fxml.FXMLLoader.loadImpl(FXMLLoader.java:2548)
	at javafx.fxml.FXMLLoader$IncludeElement.constructValue(FXMLLoader.java:1156)
	at javafx.fxml.FXMLLoader$ValueElement.processStartElement(FXMLLoader.java:756)
	at javafx.fxml.FXMLLoader.processStartElement(FXMLLoader.java:2808)
	at javafx.fxml.FXMLLoader.loadImpl(FXMLLoader.java:2634)
	at javafx.fxml.FXMLLoader.loadImpl(FXMLLoader.java:2548)
	at javafx.fxml.FXMLLoader$IncludeElement.constructValue(FXMLLoader.java:1156)
	at javafx.fxml.FXMLLoader$ValueElement.processStartElement(FXMLLoader.java:756)
	at javafx.fxml.FXMLLoader.processStartElement(FXMLLoader.java:2808)
	at javafx.fxml.FXMLLoader.loadImpl(FXMLLoader.java:2634)
	at javafx.fxml.FXMLLoader.load(FXMLLoader.java:2532)
	at org.cryptomator.ui.common.FXMLLoaderFactory.load(FXMLLoaderFactory.java:46)
	at org.cryptomator.ui.common.FXMLLoaderFactory.createScene(FXMLLoaderFactory.java:60)
	at org.cryptomator.ui.mainwindow.MainWindowModule.provideMainScene(MainWindowModule.java:64)
	at org.cryptomator.ui.mainwindow.MainWindowModule_ProvideMainSceneFactory.provideMainScene(MainWindowModule_ProvideMainSceneFactory.java:36)
	at org.cryptomator.ui.mainwindow.MainWindowModule_ProvideMainSceneFactory.get(MainWindowModule_ProvideMainSceneFactory.java:27)
	at org.cryptomator.ui.mainwindow.MainWindowModule_ProvideMainSceneFactory.get(MainWindowModule_ProvideMainSceneFactory.java:10)
	at dagger.internal.DoubleCheck.get(DoubleCheck.java:47)
	at org.cryptomator.ui.mainwindow.MainWindowComponent.showMainWindow(MainWindowComponent.java:28)
	at org.cryptomator.ui.fxapp.FxApplication.lambda$showMainWindow$1(FxApplication.java:107)
	at com.sun.javafx.application.PlatformImpl.lambda$runLater$10(PlatformImpl.java:428)
	at java.base/java.security.AccessController.doPrivileged(AccessController.java:391)
	at com.sun.javafx.application.PlatformImpl.lambda$runLater$11(PlatformImpl.java:427)
	at com.sun.glass.ui.InvokeLaterDispatcher$Future.run(InvokeLaterDispatcher.java:96)
	at com.sun.glass.ui.gtk.GtkApplication._runLoop(Native Method)
	at com.sun.glass.ui.gtk.GtkApplication.lambda$runLoop$11(GtkApplication.java:277)
	at java.base/java.lang.Thread.run(Thread.java:832)
Caused by: java.lang.RuntimeException: Cannot Resolve Session Bus Address
	at org.freedesktop.dbus.connections.impl.DBusConnection.lambda$getConnection$1(DBusConnection.java:257)
	at org.freedesktop.dbus.connections.impl.DBusConnection.getConnection(DBusConnection.java:165)
	at org.freedesktop.dbus.connections.impl.DBusConnection.getConnection(DBusConnection.java:228)
	at org.freedesktop.dbus.connections.impl.DBusConnection.getConnection(DBusConnection.java:184)
	at org.freedesktop.secret.simple.SimpleCollection.<clinit>(SimpleCollection.java:39)
	... 58 more`

**Comment by infeo** [comment]
I opened an issue in the upstream dbus-java library regarding the RuntimeException:
https://github.com/hypfvieh/dbus-java/issues/128

The next release contains a change such that the static methods do not throw it anymore :D @swiesend @purejava It might be worthwile checking, if that version can be used in your libraries but the maintainer mentions possible breaking changes.
